server/src/diagnostic_ext.rs

```diff
@@ -0,0 +1,80 @@
+// Copyright (c) ZeroC, Inc.
+
+use slicec::diagnostics::{Diagnostic, DiagnosticLevel, Note};
+use tower_lsp::lsp_types::{
+    CodeDescription, DiagnosticRelatedInformation, Location, NumberOrString, Position, Range, Url,
+};
+
+pub fn try_into_lsp_diagnostic(
+    diagnostic: &Diagnostic,
+) -> Option<tower_lsp::lsp_types::Diagnostic> {
+    let severity = match diagnostic.level() {
+        DiagnosticLevel::Error => Some(tower_lsp::lsp_types::DiagnosticSeverity::ERROR),
+        DiagnosticLevel::Warning => Some(tower_lsp::lsp_types::DiagnosticSeverity::WARNING),
+        DiagnosticLevel::Allowed => None,
+    };
+
+    // Map the spans to ranges, if span is none, return none
+    let range = match diagnostic.span() {
+        Some(span) => {
+            let start = tower_lsp::lsp_types::Position::new(
+                (span.start.row - 1) as u32,
+                (span.start.col - 1) as u32,
+            );
+            let end = tower_lsp::lsp_types::Position::new(
+                (span.end.row - 1) as u32,
+                (span.end.col - 1) as u32,
+            );
+            Range::new(start, end)
+        }
+        None => return None,
+    };
+
+    let message = diagnostic.message();
+    let related_information: Option<Vec<DiagnosticRelatedInformation>> = Some(
+        diagnostic
+            .notes()
+            .iter()
+            .filter_map(|n| try_into_lsp_diagnostic_related_information(n))
+            .collect(),
+    );
+
+    Some(tower_lsp::lsp_types::Diagnostic {
+        range,
+        severity,
+        code: Some(NumberOrString::String(diagnostic.code().to_owned())),
+        code_description: Some(CodeDescription { href:
+                // Create a URL object to https://docs.icerpc.dev
+                Url::parse("https://docs.icerpc.dev").unwrap(),
```

Comment:
I would use `None`, I assume this is supposed to provide information about the specific error code, pointing the user to the main doc site isn't very helpful.

We can later add a link if we have the error code descriptions in the docs.

Reply:
Yes this was just from testing good catch!